### PR TITLE
fix(presets): `helpers:pinGitHubActionDigestsToSemver` optional `v` versioning prefix

### PR DESCRIPTION
**Pull request by MPV**
## Changes

Optional `v` prefix in versioning regex in `helpers:pinGitHubActionDigestsToSemver`.

## Context

A follow-up on:
- https://github.com/renovatebot/renovate/pull/23663

...after the changes made to versioning recently, possibly in:
- https://github.com/renovatebot/renovate/issues/27317

...to avoid/fix the issues seen in:
- https://github.com/renovatebot/renovate/discussions/27429

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/presets/internal/helpers.ts [hunk 1]
```diff
@@ -37,7 +37,7 @@ export const presets: Record<string, Preset> = {
         extends: ['helpers:pinGitHubActionDigests'],
         extractVersion: '^(?<version>v\\d+\\.\\d+\\.\\d+)$',
         versioning:
-          'regex:^v(?<major>\\d+)(\\.(?<minor>\\d+)\\.(?<patch>\\d+))?$',
+          'regex:^v?(?<major>\\d+)(\\.(?<minor>\\d+)\\.(?<patch>\\d+))?',
```
Comment (rarkins) [inline lib/config/presets/internal/helpers.ts:40]: Is dropping the $ necessary?

Comment (viceice) [reply lib/config/presets/internal/helpers.ts:40]: I don't think so

Comment (viceice) [reply lib/config/presets/internal/helpers.ts:40]:
```suggestion
          'regex:^v?(?<major>\\d+)(\\.(?<minor>\\d+)\\.(?<patch>\\d+))?$',
```

Comment (MPV) [reply lib/config/presets/internal/helpers.ts]: `$` re-added. I just added it since it was a part of your suggestion in the linked discussion.

**Review by rarkins — APPROVED** [review]

**Pull request merged**